bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/metadata/RuntimeTypeAdapterFactory.java

```diff
@@ -0,0 +1,478 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+/*
+ * Copyright (C) 2011 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copied from
+ * https://raw.githubusercontent.com/google/gson/master/extras/src/main/java/com/google/gson/typeadapters/RuntimeTypeAdapterFactory.java
+ * and repackaged here with additional content from
+ * com.google.gson.internal.{Streams,TypeAdapters,LazilyParsedNumber}
+ * to avoid using the internal package.
+ */
```

Comment:
fix on next commit

Reply:
Just to clarify, I move the file to a ThirdParty folder.
But i was not able to remove the Openhab headers because of checkstyle verification that trigger an error on build.
So the file still have the double headers, one for openhab, and one from the initial source.
Let me know if I can do something about this.

I also modified the Notice file to add some info about this thirdparty.

Reply:
You need to move it from:
`bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/thirdparty/RuntimeTypeAdapterFactory.java`
to:
`bundles/org.openhab.binding.siemenshvac/src/3rdparty/main/java/com/google/gson/typeadapters/RuntimeTypeAdapterFactory.java`

i.e. move it back to its original name space `com.google.gson.typeadapters` and restore the original licence header. This should exclude it from the checkstyle checks, or at least the copyright header check.

Reply:
ok, thanks for the guideline.
Fix in next commit

Reply:
Sorry, I made a mistake, directory "main" in not needed (I checked other bindings):
`bundles/org.openhab.binding.siemenshvac/src/3rdparty/java/com/google/gson/typeadapters/RuntimeTypeAdapterFactory.java`